Cray compilers don't like old directives

Reported by Sarat Sreepathi

Trying to build E3SM on a Cray testbed, I ran into the following issue with a workaround. 

Issue:
```
ftn-855 crayftn: ERROR M_ATTRVECT, File = ../../../home/users/coe0149/repos/E3SM/cime/src/externals/mct/mct/m_AttrVect.F90, Line = 67, Column = 9 
  The compiler has detected errors in module "M_ATTRVECT".  No module information file will be created for this module.

ftn-1381 crayftn: ERROR EXPORTIATTR_, File = ../../../home/users/coe0149/repos/E3SM/cime/src/externals/mct/mct/m_AttrVect.F90, Line = 1952 
  A statement must not appear between the COLLAPSE directive and the DO loop that follows it. 

Cray Fortran : Version 9.1.3 (20200225214845_454c23bd9080e42515651f42b36ba77db1022811)
Cray Fortran : Compile time:  0.1501 seconds
Cray Fortran : 4138 source lines
Cray Fortran : 2 errors, 0 warnings, 0 other messages, 0 ansi
Cray Fortran : "explain ftn-message number" gives more information about each message.
gmake[1]: *** [m_AttrVect.o] Error 1
```

Workaround: Removing two lines with  **!DIR$ COLLAPSE**
I presume they are expected to be commented out but they trigger the Cray compiler.
```
diff --git a/cime/src/externals/mct/mct/m_AttrVect.F90 b/cime/src/externals/mct/mct/m_AttrVect.F90
index d186fb8..82931d1 100644
--- a/cime/src/externals/mct/mct/m_AttrVect.F90
+++ b/cime/src/externals/mct/mct/m_AttrVect.F90
@@ -845,7 +845,6 @@
   if(myZeroInts) then ! zero out INTEGER attributes
      if(List_allocated(aV%iList)) then
         if(associated(aV%iAttr) .and. (nIAttr_(aV)>0)) then
-!DIR$ COLLAPSE
            aV%iAttr=0
         endif
      endif
@@ -854,7 +853,6 @@
   if(myZeroReals) then ! zero out REAL attributes
      if(List_allocated(aV%rList)) then
         if(associated(aV%rAttr) .and. (nRAttr_(aV)>0)) then
-!DIR$ COLLAPSE
            aV%rAttr=0._FP
         endif
```

Related issue: https://github.com/E3SM-Project/E3SM/issues/3509
Please let me know if this issue needs to be replicated in this repo.

There was a similar line (albeit lowercase) in m_MatAttrVectMul.F90 that did not trigger the above problem.

--- a/mct/m_AttrVect.F90
+++ b/mct/m_AttrVect.F90
@@ -845,7 +845,6 @@
   if(myZeroInts) then ! zero out INTEGER attributes
      if(List_allocated(aV%iList)) then
         if(associated(aV%iAttr) .and. (nIAttr_(aV)>0)) then
-!DIR$ COLLAPSE
            aV%iAttr=0
         endif
      endif
@@ -854,7 +853,6 @@
   if(myZeroReals) then ! zero out REAL attributes
      if(List_allocated(aV%rList)) then
         if(associated(aV%rAttr) .and. (nRAttr_(aV)>0)) then
-!DIR$ COLLAPSE
            aV%rAttr=0._FP
         endif
      endif

